### PR TITLE
[Addon]: add envoy-gateway to experiment

### PR DESCRIPTION
**Pull request by Xunzhuo**
Signed-off-by: bitliu <bitliu@tencent.com>

### Description of your changes

Fixes: https://github.com/kubevela/catalog/issues/587

### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [x] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [x] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [x] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: experimental/addons/envoy-gateway/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# envoy-gateway
+
+Envoy Gateway is an open source project for managing Envoy Proxy as a standalone or Kubernetes-based application gateway.
```
Comment (wonderflow) [inline experimental/addons/envoy-gateway/README.md:3]: Can you help add some usage docs here? Users need the basic knowledge about how to use this addon.

And please also add some application examples.

**Comment by wangyikewxgm** [comment]
Thanks for your contribution!

File: experimental/addons/envoy-gateway/README.md (hunk 1 above)
Comment (Xunzhuo) [reply experimental/addons/envoy-gateway/README.md:3]: Thanks, I will address this comment ASAP.

File: experimental/addons/envoy-gateway/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+```shell
+kubectl wait --timeout=5m -n envoy-gateway-system deployment/envoy-gateway --for=condition=Available
+```
+
+Install the GatewayClass, Gateway, HTTPRoute and example app:
+
+```shell
+kubectl apply -f https://github.com/envoyproxy/gateway/releases/download/latest/quickstart.yaml
+```
+
```
Comment (wonderflow) [inline experimental/addons/envoy-gateway/README.md:18]: Can you provide some samples about how KubeVela Application can use it?

I think we need to define some traits, right?

Comment (Xunzhuo) [reply experimental/addons/envoy-gateway/README.md:18]: This requires some refactors to gateway-trait and other gwapi implementations. Not sure if I should do whole things in a single PR or separated them.

**Review by wonderflow — APPROVED** [review]

File: experimental/addons/envoy-gateway/README.md (hunk 2 above)
Comment (Xunzhuo) [reply experimental/addons/envoy-gateway/README.md:18]: Sure

**Pull request merged**